Draft of the H2 roadmap

jrainville opened the pull request:
Please leave comments on what should be moved around, what should be done in later milestones, estimates, etc.

(again pro tip: check in preview mode, especially for the graph)

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 168782a6 | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-18108/1/) | 2025-06-10 20:15:29 | ~7 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-18108/1//consoleText) |

alexjba commented:
The dApp browser is a really nice addition! Let's make it mobile compatible (to use OS webviews instead of the webengine). Another advantage here is that the app could potentially become App Store compliant for MacOs as well.

noeliaSD reviewed (approved):
LGTM! Just one comment about the `UI modularization`, the current epic doesn't include all the needed effort. There are some steps to improve on wallet section and the chat domain is the most broken down but still not everything covered.

caybro commented on docs/roadmap.md:
```diff
@@ -0,0 +1,69 @@
+
+Epic: https://github.com/status-im/status-desktop/issues/18029
+
+Estimated release: End of August
```
A bit too tight schedule considering only month after 2.35 _and_ the summer vacations period

jrainville commented:
Thanks everyone for reviewing. I updated according to your comments.

I also improved the graph by adding the milestone releases estimated dates and I added File uploads on Codex to the 2.37 milestone.

The last big thing I added is FURPS for all written Epic!
They are in the `docs/FURPS` folder.

Some of them I just grabbed from the issues (kudos @alexjba). The others I generated using an AI from the epic and designs, then I edited them.
Finally, I linked the roadmap to the FURPS.

  When this PR is merged, we can link the Epics to the right FURPS document

noeliaSD commented on docs/FURPS/jump-to-screen-shell.md:
```diff
@@ -0,0 +1,28 @@
+## Functionality
+- Support navigation to communities, chats, accounts, or dApps through a unified launcher interface.
+- Display recent items with live metadata (e.g., unread count, media preview, timestamps).
+- Order from most recent. When item is clicked, it moves to the top.
```
I already expressed my mixed feelings on the PR related to this behavior: `It's a bit annoying to me that the grid order changes each time I click a new item. It's nice to have the most recents on top but then it makes it difficult to find a specific one since it's never at the same position. Maybe we could have the possibility of ordering by recents or not, so user can decide`

jrainville replied:
Good point. I don't have hard feelings either way. 

I hope more people can chime in and express their opinion so we can decide if we drop that functionality

caybro replied:
Yeah... this was requested by Ben; I'm also not very keen on the items jumping around; I'd much rather see them ordered either by "notifications" first, and then alphabet

noeliaSD commented on docs/FURPS/jump-to-screen-shell.md:
Also commented it on the PR: `Would it make sense to add the Sign out & Quit option in this view too?`

caybro replied:
Is is there; want to get rid of it?

![image](https://github.com/user-attachments/assets/e7811a9b-aa1e-48a4-a774-60b430c36e1c)

alaibe commented on docs/roadmap.md:
```diff
@@ -0,0 +1,69 @@
+- [QT6 migration](https://github.com/status-im/status-desktop/issues/17622)
+- [Tablet Build](https://github.com/status-im/status-desktop/issues/17941)
+- [Jump to screen (Shell)](https://github.com/status-im/status-desktop/issues/17971)
+- [Memory improvements](https://github.com/status-im/status-go/issues/6544)
```
yes I wouldn't expect us to be able to work on all the rest correctly, maybe smart the mem improvement but aim to finish more in 2.36

86doteth commented:
also im wondering where keycard shell integration work will be. maybe its estimated to require so little work that it doesnt need to be mentioned here but i hope that assumption is well reasoned. last i heard shell should start shipping in q4 and i think its crucial shell users can immediately be funneled into status wallet since they will probably be most important for revenue and hopefully even formation of communities of snt whales

jrainville commented:
> also im wondering where keycard shell integration work will be. maybe its estimated to require so little work that it doesnt need to be mentioned here but i hope that assumption is well reasoned. last i heard shell should start shipping in q4 and i think its crucial shell users can immediately be funneled into status wallet since they will probably be most important for revenue and hopefully even formation of communities of snt whales

Good catch, I forgot to add it. Indeed we will want to support Shell!

The Q4 roadmap is a bit hazy because we try to adapt to feedback.

I'll add Shell integration! In theory it shouldn't be too hard since it's only about supporting ERC-4527 (knocks on wood)

caybro commented on docs/FURPS/jump-to-screen-shell.md:
```diff
@@ -0,0 +1,28 @@
+# Jump to screen (Shell) FURPS ([#17971](https://github.com/status-im/status-desktop/issues/17971))
```
```suggestion
# Jump to screen (Home Page) FURPS ([#17971](https://github.com/status-im/status-desktop/issues/17971))
```

86doteth commented:
> > also im wondering where keycard shell integration work will be. maybe its estimated to require so little work that it doesnt need to be mentioned here but i hope that assumption is well reasoned. last i heard shell should start shipping in q4 and i think its crucial shell users can immediately be funneled into status wallet since they will probably be most important for revenue and hopefully even formation of communities of snt whales
> 
> Good catch, I forgot to add it. Indeed we will want to support Shell!
> 
> The Q4 roadmap is a bit hazy because we try to adapt to feedback.
> 
> I'll add Shell integration! In theory it shouldn't be too hard since it's only about supporting ERC-4527 (knocks on wood)

yeah and also the authentication and signing popups that now ask for pins should recognize when a shell is used so pins and signing actions happen on shell and are properly received and displayed by status wallet. 

and it should work over usb as well imo since not everyone has webcams available or open at all times for privacy or convenience reasons.

jrainville commented:
> could we squeeze in somewhere the keycard support for the new mobile app? Although I think it's going to require a big effort. We might need a new keycard lib if we can't nicely replace the libpcsc in the go lib - assuming we won't have the same issue we have on Intel Macs where we can't use both keycard and status-go libs.

Yes we can add a new task. I had assumed it would be part of the Mobile Epic itself, but if it's a way bigger issue, we can track it separately.

alexjba commented:
> > could we squeeze in somewhere the keycard support for the new mobile app? Although I think it's going to require a big effort. We might need a new keycard lib if we can't nicely replace the libpcsc in the go lib - assuming we won't have the same issue we have on Intel Macs where we can't use both keycard and status-go libs.
> 
> Yes we can add a new task. I had assumed it would be part of the Mobile Epic itself, but if it's a way bigger issue, we can track it separately.

It is somewhat part of the mobile epic - as a nice to have for the initial roll-out, but it deserves its own space for sure.
